src/Microsoft.Owin.Security.Authorization/IAuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public interface IAuthorizationDependenciesProvider
+    {
+        Func<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnCreate { get; }
```

Comment:
Instead of a `Func<>`, this would be simpler as a plain old method:
```cs
AuthorizationDependencies Create(AuthorizationOptions options, IOwinContext context);
```